src/dsl.ts

```diff
@@ -637,86 +655,152 @@ const permission = seqMap(
   },
 );
 
-const relation = seqMap(
+const relation: Parser<ParsedRelation> = seqMap(
   index,
   seq(
     lexeme(string("relation")),
     identifier,
     colon.then(typeExpr).skip(terminator.atMost(1)),
   ),
   index,
-  function (startIndex, data, endIndex) {
+  function (startIndex, [_, name, allowedTypes], endIndex) {
     return {
       kind: "relation",
-      name: data[1],
-      allowedTypes: data[2],
-      range: { startIndex: startIndex, endIndex: endIndex },
+      name,
+      allowedTypes,
+      range: { startIndex, endIndex },
```

Comment:
I did a few of these drive-by refactors. Using array destructuring and object shorthand notation cleans things up in my eyes.